Managers/AudioMan.h

```diff
@@ -1,707 +1,446 @@
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Sets/updates the frequency/pitch for the music channel.
-// Arguments:       New pitch, a multiplier of the original normal frequency. Keep it > 0
-// Return value:    Whether a sample of the Sound is currently being played by any of the
-//                  channels, and the pitch was successfully set.
-
-    bool SetMusicPitch(float pitch);
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          IsPlaying
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Reports whetehr a certain Sound's last played sample is being played
-//                  currently.
-// Arguments:       A pointer to a Sound object. Ownership IS NOT transferred!
-// Return value:    Whether the LAST sample that was played of the Sound is currently being
-//                  played by any of the channels.
-
-    bool IsPlaying(Sound *pSound);
+#define g_AudioMan AudioMan::Instance()
 
+namespace RTE {
+	class Sound;
+
+	/// <summary>
+	/// The singleton manager of the WAV sound effects and OGG music playback.
+	/// </summary>
+	class AudioMan : public Singleton<AudioMan> {
+
+	public:
+		CLASSINFOGETTERS
+
+		enum PlaybackPriority {
+			PRIORITY_LOW = 0,
+			PRIORITY_HIGH = 100,
+			PRIORITY_NOATTENUATION,
+			PRIORITY_COUNT
+		};
+
+		enum NetworkSoundState {
+			SOUND_PLAY = 0,
+			SOUND_STOP,
+			SOUND_SET_PITCH,
+			SOUND_SET_ATTENUATION
+		};
+
+		struct NetworkSoundData {
+			unsigned char State;
+			size_t SoundHash;
+			unsigned char AffectedByPitch;
+			short int Distance;
+			short int Channel;
+			short int Loops;
+			float Pitch;
+		};
+
+		enum NetworkMusicState {
+			MUSIC_PLAY = 0,
+			MUSIC_STOP,
+			MUSIC_SILENCE,
+			MUSIC_SET_PITCH
+		};
+
+		struct NetworkMusicData {
+			unsigned char State;
+			char Path[256];
+			int Loops;
+			double Position;
+			float Pitch;
+		};
+
+#pragma region Creation
+		/// <summary>
+		/// Constructor method used to instantiate a AudioMan object in system memory.
+		/// Create() should be called before using the object.
+		/// </summary>
+		AudioMan() { Clear(); }
+
+		/// <summary>
+		/// Makes the AudioMan object ready for use.
+		/// </summary>
+		/// <returns>An error return value signaling success or any particular failure. Anything below 0 is an error signal.</returns>
+		virtual int Create();
+#pragma endregion
+
+#pragma region INI Handling
+		/// <summary>
+		/// Saves the complete state of this AudioMan to an output stream for later recreation with Create(Reader &reader);
+		/// </summary>
+		/// <param name="writer">A Writer that the AudioMan will save itself with.</param>
+		/// <returns>An error return value signaling success or any particular failure. Anything below 0 is an error signal.</returns>
+		virtual int Save(Writer &writer) const { return 0; }
```

Comment:
I believe this is junk. We'd never be writing audioman to ini (which I'm pretty sure is what these are for). Honestly we should really get rid of all of them, they're legacy from when I guess it was useful to be able to write a class to ini but I see no benefit to them ever nowadays.